operator/controllers/kyma_controller.go

```diff
@@ -375,52 +376,61 @@ func (r *KymaReconciler) CreateOrUpdateModules(ctx context.Context, kyma *v1alph
 		outdatedOverride := kyma.HasOutdatedOverride(name)
 
 		if k8serrors.IsNotFound(errors.Unwrap(err)) {
-			logger.Info("module not found, attempting to create it...")
-			err := r.CreateModule(ctx, name, kyma, module)
-			if err != nil {
-				return false, err
-			}
-
-			status.Helper(r).SyncReadyConditionForModules(kyma, parsed.Modules{name: module},
-				v1alpha1.ConditionStatusFalse, fmt.Sprintf("initial condition for %s module CR", module.Name))
-
-			logger.Info("successfully created module CR")
-
-			if outdatedOverride {
-				kyma.RefreshOverride(name)
-			}
-
-			return true, nil
-		}
-
-		update := func() (bool, error) {
-			if err := r.UpdateModule(ctx, name, kyma, module); err != nil {
-				return false, err
-			}
-			if outdatedOverride {
-				kyma.RefreshOverride(name)
-			}
-			logger.Info("successfully updated module CR")
-			status.Helper(r).SyncReadyConditionForModules(kyma, parsed.Modules{name: module},
-				v1alpha1.ConditionStatusFalse, "updated condition for module CR")
-			return true, nil
+			return r.createAndUpdateNotFoundModule(ctx, kyma, module, logger, outdatedOverride, name)
```

Comment:
Lint issue fixed with proper refactoring in https://github.com/kyma-project/kyma-operator/pull/84